bpf-notifier/notifier.bpf.c

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+/* Copyright (c) 2023 Red Hat */
+
+#include <linux/bpf.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_endian.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_core_read.h>
+#include <linux/if.h>
+#include <linux/if_ether.h>
+
+#include "notifier.h"
+
+struct net_device {
+	char name[IFNAMSIZ];
+} __attribute__((preserve_access_index));
+
+struct switchdev_notifier_info {
+  struct net_device *dev;
+  struct netlink_ext_ack *extack;
+  const void *ctx;
+} __attribute__((preserve_access_index));
+
+struct switchdev_notifier_fdb_info {
+	struct switchdev_notifier_info info;
+	const unsigned char *addr;
+	__u16 vid;
+	__u8 added_by_user:1,
+	     is_local:1,
+	     locked:1,
+	     offloaded:1;
+} __attribute__((preserve_access_index));;
+
+struct netlink_ext_ack {
```

Comment:
Is it possible to include netlink.h for this struct definition?

Reply:
I could because it is UAPI, but I don't use the param so I chose to write an empty definition for the CO-RE magic. Hmm, it turns out that I can just omit the *extack parameter from the function signature and the bpf parameter binding magic still works.